Bullet-dodging game/Bullet-Dodging Game in Processing.md

```diff
@@ -0,0 +1,2191 @@
+
+```
+class Player {
+  int size = 100;
+  int xpos = int(random(size, width - size));
+  int ypos = 1000;
+  
+  void update() {
+    if (keyPressed) {
+      if (keyCode == LEFT) {
+        xpos -= 10;
+      } else if (keyCode == RIGHT) {
+        xpos += 10;
+      } else if (keyCode == UP) {
+        ypos -= 10;
+      } else if (keyCode == DOWN) {
+        ypos += 10;
+      }
+    }
+    // prevent the player from moving off-screen
+    xpos = constrain(xpos, size / 2, width - size / 2);
+  }
+
+  void show() {
+    fill(0);  // filled with black
+    circle(xpos, ypos, size);
+  }
+}
+```
+
+We can now declare an instance `Player player;` at the top of the code as a global variable and initialize it in `setup()`: `  player = new Player();`. Then by calling `player.update();player.show();` in `draw()`, we should be able to control the player's movement now.
+
+
+
+Likewise, we can also define a Bullet class with functions `update()`, `show()` and `check_collision()`. The bullets will drop from top to bottom (vertical position increases from 0 to height) at a speed of 5 and at the same time will check if they have hit the player.
+
+```
+class Bullet {
+  int size = 10;
+  int xpos = int(random(size, width - size));
+  int ypos;
+  
+  Bullet(int ypos) {
+    this.ypos = ypos;
+  }
+
+  void update() {
+    ypos += 5;
+    if (ypos > height + size / 2) {  // reuse the bullets
+      xpos = int(random(size, width - size));
+      ypos = - size / 2;
+    }
+  }
+
+  void show() {
+    fill(255);
+    stroke(255, 0, 0);
+    strokeWeight(5);
+    circle(xpos, ypos, size);
+    noStroke();
+  }
+
+  boolean check_collision() {
+    if (dist(xpos, ypos, player.xpos, player.ypos) < (size + player.size) / 2) {
+      return true;
+    }
+    return false;
+  }
+}
+```
+
+We can now declare an array of Bullet at the top of the code as global variables:
+
+```
+int MAX_BULLET = 25;
+Bullet bullets[] = new Bullet[MAX_BULLET];
+```
+
+Let's initialize the bullets with varying initial vertical positions in `setup()`:
+
+```
+for (int i = 0; i < MAX_BULLET; i++) {
+  bullets[i] = new Bullet(- i * height / MAX_BULLET);
+}
+```
+
+Then we can `update()`, `show()` and `check_collision()` in `draw()`:
+
+```
+for (int i = 0; i < MAX_BULLET; i++) {
+  bullets[i].update();
+  bullets[i].show();
+  if (bullets[i].check_collision()) {
+    exit();
+  }
+}
+```
+
+We haven't dealt with the case when the bullet hits the player, but at least we already have a rather primitive version of the game to play now.
+
+
+
+## Part III: Adding a Life and Scoring System
+
+Without a second chance, however, the player in our initial version of the game seems prone to death. So let's add a life system to make the game more enjoyable.

```

Comment:
weird sentence to start a section. something like, "we want to allow the player to have multiple chances in our game, otherwise..."